File: src/my_project/easy_problems/from101to150/unique_email_address.py
from typing import List 

class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:

        def split_concat_string(email: str) -> str:
            temp_list: List[str]= list()
            temp_list = email.split("@")
            str_left = temp_list[0]
            str_right = temp_list[-1]
            temp_list_local_name: List[str] = list()
            temp_list_local_name = str_left.split("+")
            str_left = temp_list_local_name[0]
            temp_list_local_name = str_left.split(".")
            str_left = "".join(temp_list_local_name)

            return str_left +"@"+ str_right

        answer_list: List[str] = list()

        for email in emails: 
            answer_list.append(split_concat_string(email))

        
        return len(set(answer_list))

solution = Solution()

print(solution.numUniqueEmails(emails = ["test.email+alex@leetcode.com","test.e.mail+bob.cathy@leetcode.com","testemail+david@lee.tcode.com"]))

print(solution.numUniqueEmails(emails = ["a@leetcode.com","b@leetcode.com","c@leetcode.com"]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/uniqueEmailAddress.ts
function numUniqueEmails(emails: string[]): number {

    function splitConcatString(email: string) : string{

        let tempList: Array<string> = []
        tempList = email.split("@")
        let strLeft = tempList[0]
        let strRight = tempList[tempList.length-1]
        let tempListLocalName: Array<string> = strLeft.split("+")
        strLeft = tempListLocalName[0]
        tempListLocalName = strLeft.split(".")
        strLeft = tempListLocalName.join("")
        
        return strLeft + "@" + strRight
    }

    let answerList: Array<string> = []

    for (let email of emails){
        answerList.push(splitConcatString(email))
    }

    let answerSet: any = new Set()
    for (let email of answerList){
        answerSet.add(email)
    }


    return answerSet.size

};

console.log(numUniqueEmails(["test.email+alex@leetcode.com","test.e.mail+bob.cathy@leetcode.com","testemail+david@lee.tcode.com"]))


console.log(numUniqueEmails(["a@leetcode.com","b@leetcode.com","c@leetcode.com"]))





